packages/router/src/link.js

```diff
@@ -33,6 +30,9 @@ export function useLink( params, state, shouldReplace = false ) {
 		...Object.keys( currentArgs )
 	);
 
+	// TODO:
```

Comment:
The existing theme preview functionality has been introduced at https://github.com/WordPress/gutenberg/pull/50030 and works this way: you can add a `wp_theme_preview=THEME` URL parameter and the new theme will be previewed. It works in the site editor, post editor, and front-end.

To make sure all links respect the existing `wp_theme_preview` parameter, navigation & links should use `useLink` or `Link` utilities. Some thoughts:

- We need to account for the `wp_theme_preview` parameter in more places than the site editor. For example, post actions, such as "View" (takes you to front-end) should respect that parameter. These actions are defined in the `@wordpress/editor` package so they can be shared between edit-post & edit-site. How do we go about making this work? This PR is a potential approach, but I'd like to double-check if it's sensible to go this way ([see](https://github.com/WordPress/gutenberg/pull/63849/files#r1687853902)).

- @jsnajdr is it problematic that the params maintain the `wp_theme_preview` but it's not present in the history? Haven't been able to dig down into the router/history behavior.